statement.go

```diff
@@ -25,6 +25,11 @@ func (b StatementBuilderType) Delete(from string) DeleteBuilder {
 	return DeleteBuilder(b).From(from)
 }
 
+// Case returns a CaseBuilder for this StatementBuilderType.
+func (b StatementBuilderType) Case() CaseBuilder {
+	return CaseBuilder(b)
```

Comment:
I don't think this method is necessary since CaseBuilder doesn't use PlaceholderFormat or RunWith (the only things StatementBuilder really provides). You should be able to just have a Case function that returns either `CaseBuilder(StatementBuilder)` or just `CaseBuilder(builder.EmptyBuilder)`.